metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/collect/ProtocolHealthCheckStrategy.java

```diff
@@ -149,4 +155,42 @@ public void checkHttpHealth(String hostname, Monitor hostMonitor, MetricFactory
 			telemetryManager.getStrategyTime()
 		);
 	}
+
+	/**
+	 * Check SNMP protocol health on the hostname for the host monitor
+	 * Criteria: SNMP Get Next on '1.3.6.1' SNMP OId must be successful.
+	 *
+	 * @param hostMonitor   An endpoint host monitor
+	 * @param hostname      The hostname on which we perform health check
+	 * @param metricFactory The metric factory used to collect the health check metric
+	 */
+	public void checkSnmpHealth(String hostname, Monitor hostMonitor, MetricFactory metricFactory) {
+		String snmpResult = null;
+		// Retrieve SNMP Configuration from the telemetry manager host configuration
+		final SnmpConfiguration snmpConfiguration = (SnmpConfiguration) telemetryManager
+			.getHostConfiguration()
+			.getConfigurations()
+			.get(SnmpConfiguration.class);
+		// Stop the SNMP health check if there is not an SNMP configuration
+		if (snmpConfiguration == null) {
+			return;
+		}
+		// Execute SNMP test command
+		try {
+			snmpResult = clientsExecutor.executeSNMPGetNext(SNMP_OID, snmpConfiguration, hostname, true);
+		} catch (Exception e) {
+			log.debug(
+				"Hostname {} - Checking SNMP protocol status. SNMP exception when performing a test SNMP Get Next: ",
+				hostname,
+				e
+			);
+		}
+		// Generate a metric from the Snmp result
+		metricFactory.collectNumberMetric(
+			hostMonitor,
+			String.format(UP_METRIC_FORMAT, "SNMP"),
+			snmpResult != null ? UP : DOWN,
+			telemetryManager.getStrategyTime()

```

Comment:
We know the `strategyTime` (member of this class), we shouldn't call `telemetryManager.getStrategyTime()`.